libs/coin-modules/coin-tezos/src/api/index.ts

```diff
@@ -82,46 +143,121 @@ async function craft(
     storageLimit: fees.parameters?.storageLimit?.toString(),
   }));
 
-  const { contents } = await craftTransaction(
-    { address: transactionIntent.sender },
-    {
-      type: transactionIntent.type,
-      recipient: transactionIntent.recipient,
-      amount: transactionIntent.amount,
-      fee,
-    },
-  );
+  // map generic staking intents to deal with tezos operation modes (delegate/undelegate)
+  const mappedType = mapIntentTypeToTezosMode(transactionIntent.type);
+
+  // guard: send max is incompatible with delegated accounts
+  if (mappedType === "send" && transactionIntent.useAllAmount) {
+    const senderInfo = await api.getAccountByAddress(transactionIntent.sender);
+    if (senderInfo.type === "user" && senderInfo.delegate?.address) {
+      throw new RecommendUndelegation();
+    }
+  }
+
+  // compute amount to use in forge
+  let amountToUse = transactionIntent.amount;
+  if (mappedType === "send" && transactionIntent.useAllAmount) {
+    const senderInfo = await api.getAccountByAddress(transactionIntent.sender);
+    if (senderInfo.type === "user") {
+      const bal = BigInt(senderInfo.balance);
+      const feeBI = BigInt(fee.fees || "0");
+      amountToUse = bal > feeBI ? bal - feeBI : 0n;
+    } else {
+      amountToUse = 0n;
+    }
+  }
+
+  const accountForCraft = {
+    address: transactionIntent.sender,
+    // craftTransaction expects the current counter, it will increment internally per content
+    counter:
+      typeof transactionIntent.sequence === "number" ? transactionIntent.sequence - 1 : undefined,
+  };
+  const senderApiAcc = await api.getAccountByAddress(transactionIntent.sender);
+  const needsReveal = senderApiAcc.type === "user" && !senderApiAcc.revealed;
+  const totalFeeBI = BigInt(fee.fees || "0");
+  let txFeeBI: bigint;
+  if (needsReveal) {
+    if (totalFeeBI > BigInt(DEFAULT_FEE.REVEAL)) {
+      txFeeBI = totalFeeBI - BigInt(DEFAULT_FEE.REVEAL);
+    } else {
+      txFeeBI = 0n;
+    }
+  } else {
+    txFeeBI = totalFeeBI;
+  }
```

Comment:
I thought you would have done something like this.
```suggestion
 const totalFee = Number(fee.fees || "0");
 const txFee = needsReveal ? Math.min(totalFee - DEFAULT_FEE.REVEAL, 0) : totalFee;
```
I don't understand the need of `bigint`, only to use `toString` on it a few lines after